server/src/main/java/io/crate/planner/consumer/UpdatePlanner.java

```diff
@@ -129,12 +132,29 @@ private static Plan plan(DocTableRelation docTable,
                              Symbol query,
                              PlannerContext plannerCtx,
                              @Nullable List<Symbol> returnValues) {
-        EvaluatingNormalizer normalizer = EvaluatingNormalizer.functionOnlyNormalizer(plannerCtx.nodeContext());
+        AtomicBoolean requiresAdditionalFilteringOnNonDocKeyColumns = new AtomicBoolean(false);
```

Comment:
Thx! could we extract those blocks (also in `Get`) into a common util method?

Reply:
Yes please. Additionally, but probably very minor, we may want to use `boolean[]` in favour of the `AtomicBoolean` as I think it's cheaper.